### ISSUE TITLE
Serious problem with flows started by a message - missing and/or duplicate execution

### ISSUE THREAD

**Issue by maio** [camunda-cloud/zeebe#8140]
**Describe the bug**

Flows which are started by a message behave in strange/incorrect way when messages are published concurrently. Some messages are ignored and for some flows are executed multiple times.

**To Reproduce**

![image](https://user-images.githubusercontent.com/18138/140281725-6ad76b9d-5e6b-48aa-a8cc-54ceafb15db6.png)

1. Create flow with is triggered by message - e.g. message `foobar`.
2. In parallel (e.g. from multiple Zeebe clients) publish 10 `foobar` messages with different correlation keys and different variables.

**Expected behavior**

For each message flow is started (i.e. 10 flows) and it contains matching variables.

**Actual behavior**

Some flows are started multiple times with same variables and some messages are ignored (i.e. there is no flow executed with matching variables).

**Log/Stacktrace**

[5a47d3d38b05.log](https://github.com/camunda-cloud/zeebe/files/7473378/5a47d3d38b05.log)

***Sample code (Kotlin)***

        val base = 2700000

        runBlocking {
            repeat(10) { idInt ->
                val id = (base + idInt).toString()
                println("Sending $id")

                launch(Dispatchers.IO) {
                    val client = ZeebeClient
                        .newClientBuilder().gatewayAddress("127.0.0.1:50035")
                        .usePlaintext()
                        .build()

                    client.newPublishMessageCommand()
                        .messageName("foobar")
                        .correlationKey(id)
                        .variables(mapOf("key" to "value_$id"))
                        .send()
                        .join()
                }
            }
        }

Produces following output:
  Sending 2700000
  Sending 2700001
  Sending 2700002
  Sending 2700003
  Sending 2700004
  Sending 2700005
  Sending 2700006
  Sending 2700007
  Sending 2700008
  Sending 2700009

Flows with following variables are executed (i.e. flow is executed twice with message containing variable value_2700002, but never executed with value_2700000, value_2700001, ...):
  2700002
  2700002
  2700004
  2700004
  2700007
  2700007
  2700008
  2700008
  2700009
  2700009

**Environment:**
 - OS: Linux
 - Zeebe Version: 1.1.1

**Comment by npepinpe** [comment]
@saig0 - could it be a duplicate of https://github.com/camunda-cloud/zeebe/issues/8068 ?

**Comment by saig0** [comment]
@maio thank you for reporting. The same issue was reported recently #8068. It is fixed in the latest bugfix versions `1.1.5` / `1.2.3`.

**Comment by maio** [comment]
Thanks a lot for quick reply!

**Comment by maio** [comment]
@saig0 1.1.5 doesn't appear in releases. Would it be possible to release it with this fix please?

**Comment by oleschoenburg** [comment]
@maio We just released `1.1.5`: https://github.com/camunda-cloud/zeebe/releases/tag/1.1.5
